### PR TITLE
Refactor models

### PR DESCRIPTION
**Pull request by mawasile**
This pull request includes several changes focused on renaming DTO (Data Transfer Object) types to follow a consistent camelCase naming convention and removing an unused constant. The changes span multiple files and functions, ensuring that the new naming convention is applied uniformly across the codebase.

### DTO Renaming:

* [`internal/services/admin_management_application/api_admin_management_application.go`](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL31-R53): Updated the DTO type names from `AdminManagementApplicationDto` to `adminManagementApplicationDto` in various functions. [[1]](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL31-R53) [[2]](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL67-R63)
* [`internal/services/admin_management_application/model_admin_management_application.go`](diffhunk://#diff-d7ec1f2d636db3e57aae77982849cb09f72c997a56815b0ddffb1df65c2ee295L5-R5): Changed `AdminManagementApplicationDto` to `adminManagementApplicationDto`.
* [`internal/services/application/api_application.go`](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL37-R37): Renamed multiple DTO types, such as `EnvironmentIdDto` to `environmentIdDto` and `TenantApplicationDto` to `tenantApplicationDto`. [[1]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL37-R37) [[2]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL47-R47) [[3]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL56-R56) [[4]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL67-R67) [[5]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL77-R77) [[6]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL88-R88) [[7]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL125-R125) [[8]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL144-R144)
* [`internal/services/application/models_application.go`](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L5-R11): Updated various DTO type names to camelCase, including `TenantApplicationDto` to `tenantApplicationDto` and `EnvironmentApplicationDto` to `environmentApplicationDto`. [[1]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L5-R11) [[2]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L20-R20) [[3]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L29-R37) [[4]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L50-R116) [[5]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L125-R135)
* [`internal/services/authorization/api_user.go`](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL38-R38): Changed DTO type names from `UserDto` to `userDto` in multiple functions. [[1]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL38-R38) [[2]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL48-R56) [[3]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL70-R70) [[4]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL81-R81) [[5]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL96-R96) [[6]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL108-R108) [[7]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL151-R151) [[8]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL192-R192)

### Constant Removal:

* [`internal/services/admin_management_application/api_admin_management_application.go`](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL17-L20): Removed the unused constant `ADMIN_APP_URL`.This pull request includes several changes aimed at improving code consistency and readability by standardizing type names and removing an unnecessary constant. The most important changes include renaming DTO types to follow a consistent lowercase naming convention and removing the `ADMIN_APP_URL` constant.

### Type Renaming for Consistency:
* [`internal/services/admin_management_application/api_admin_management_application.go`](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL31-R53): Renamed `AdminManagementApplicationDto` to `adminManagementApplicationDto` and updated related method signatures and variable names accordingly. [[1]](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL31-R53) [[2]](diffhunk://#diff-d7ec1f2d636db3e57aae77982849cb09f72c997a56815b0ddffb1df65c2ee295L5-R5)
* [`internal/services/application/api_application.go`](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL37-R37): Renamed various DTO types to follow a consistent lowercase naming convention. For example, `TenantApplicationDto` to `tenantApplicationDto`, `EnvironmentApplicationDto` to `environmentApplicationDto`, and updated related method signatures and variable names accordingly. [[1]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL37-R37) [[2]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL47-R47) [[3]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL56-R56) [[4]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL67-R67) [[5]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL77-R77) [[6]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL88-R88) [[7]](diffhunk://#diff-f82ebca2ffb0f0c544c0752fa4aeee60b98312c31779004d267a573b5edc0b3dL125-R125) [[8]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L5-R11) [[9]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L20-R20) [[10]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L29-R37) [[11]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L50-R116) [[12]](diffhunk://#diff-cdcfe7e841563951fb1acf675a3b98f2313a89781e2ebb3f5bf2b70b65d038f7L125-R135)
* [`internal/services/authorization/api_user.go`](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL38-R38): Renamed `UserDto` to `userDto` and updated related method signatures and variable names accordingly. [[1]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL38-R38) [[2]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL48-R56) [[3]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL70-R70) [[4]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL81-R81) [[5]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL96-R96) [[6]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL108-R108) [[7]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL151-R151) [[8]](diffhunk://#diff-423ce31c4d46652fbe412a0f49bd085b777ce4c38e50f50613f279692259da4fL192-R192)

### Removal of Unnecessary Constants:
* [`internal/services/admin_management_application/api_admin_management_application.go`](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL17-L20): Removed the `ADMIN_APP_URL` constant and replaced its usage with the actual URL string in the `GetAdminApplication`, `RegisterAdminApplication`, and `UnregisterAdminApplication` methods. [[1]](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL17-L20) [[2]](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL31-R53) [[3]](diffhunk://#diff-ea12f9dfbad48a26f94c35cc9365d2c72cbd90b7c6c71f8149e92104b877b21dL67-R63)

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -49,3 +49,5 @@ issues:
     - "ineffectual assignment to ctx"
     - "SA4006: this value of `ctx` is never used"
     - "unused parameter (ctx|diags|req|resp)"
+    - "exported: type name will be used as environment.EnvironmentDto"
+    - "exported: type name will be used as solution.SolutionDto"
```
Comment (mattdot) [inline .golangci.yml:53]: `exported: type name will be used as ([a-zA-Z_]+)\.([A-Za-z]+)Dto`

**Review by mattdot — COMMENTED** [review]
Generally looks pretty good.  Fix broken validations and just have a few comments.  Would really like a cleaner diff because lots of these changes are already on main.

File: internal/services/authorization/models_user.go [hunk 2]
```diff
@@ -5,28 +5,28 @@ package authorization
...
 	RoleId         string `json:"roleid"`
 	Name           string `json:"name"`
 	IsManaged      bool   `json:"ismanaged"`
 	BusinessUnitId string `json:"_businessunitid_value"`
 }
 
-type SecurityRoleDtoArray struct {
-	Value []SecurityRoleDto `json:"value"`
+type securityRoleDtoArray struct {
+	Value []securityRoleDto `json:"value"`
```
[the remark this answers is not in the document]
Comment (mawasile) [reply internal/services/authorization/models_user.go:26]: I haven't seen any issues when running tests, and those are private but marshalling happens in the same package.

**Review by mattdot — CHANGES_REQUESTED** [review]

File: .golangci.yml
Comment (mattdot) [inline .golangci.yml:1]: Is there a reason that we turned off unused-parameter rule?
```yml
      - name: unused-parameter
        disabled: false
        arguments:
            - allowRegex: "(ctx|diags|req|resp)"
 ```

using the arguments might be a better approach than catch all issue exclusion at the end.

**Review by mattdot — APPROVED** [review]

**Pull request merged**